### PR TITLE
geocode shifted and working

### PR DESCRIPTION
**Pull request by mridulnagpal**
In this PR when we enter a location to input listener with default ID 'geo_location' (Can be set using options) the map pans to the entered location and the lat and lng are displayed in their respective boxes.

### CODE REVIEW

File: src/blurredLocation.js [hunk 1]
```diff
@@ -81,6 +84,8 @@ BlurredLocation = function BlurredLocation(options) {
     });
   };
 
+  panMapToGeocodedLocation(options.selector);
```
Comment (jywarren) [inline src/blurredLocation.js]: do you need to run this here? it should only be needed in the example, right? no need to run it before someone's geocoded anything, i think.

Comment (mridulnagpal) [inline src/blurredLocation.js]: sure

**Comment by mridulnagpal** [comment]
Made the changes please have a look

File: src/blurredLocation.js [hunk 2]
```diff
@@ -62,6 +61,8 @@ BlurredLocation = function BlurredLocation(options) {
     });
     var geometry = Blurred.responseJSON.results[0].geometry.location;
     options.map.setView([geometry.lat, geometry.lng],options.zoom);
+    $("#lat").val(geometry.lat);
```
Comment (jywarren) [inline src/blurredLocation.js]: Let's not do this in here - this method should only geocode. What we do with that info is an interface function. Maybe we do it either on submitting the geocode form input (ui) or when dragging the map (onDrag)?

Comment (jywarren) [inline src/blurredLocation.js]: Oh wait. I see. Here we are doing both geocoding a string and moving the map. So this is correct, maybe we just need to give a better name to the function. geocodeStringAndPan()?

But maybe the val() code should go in a callback, like a function passed to geocodeStringAndPan (string, onComplete)... What do you think?

Comment (mridulnagpal) [inline src/blurredLocation.js]: Sure, and one can input onComplete function in options and this will be the default

Comment (mridulnagpal) [inline src/blurredLocation.js]: Actually there is a different method which pans the map to geocode location namely panMapToGeocodedLocation also geocode only return the location and fills up lat and lon input listeners which can be change to some other method input by options

File: src/blurredLocation.js [hunk 3]
```diff
@@ -61,8 +67,7 @@ BlurredLocation = function BlurredLocation(options) {
     });
     var geometry = Blurred.responseJSON.results[0].geometry.location;
     options.map.setView([geometry.lat, geometry.lng],options.zoom);
```
Comment (jywarren) [inline src/blurredLocation.js]: Best move this into the onComplete method and name it onGeocodeString, maybe? But can we make it the default of the function and not the whole class? Like, `geocodeStringAndPan (string, onComplete)`, then inside the function we do `onComplete = onComplete || function onComplete (geometry) { ... }`

Comment (mridulnagpal) [inline src/blurredLocation.js]: Done, please have a look

**Comment by jywarren** [comment]
Great!

File: dist/Leaflet.BlurredLocation.js [hunk 4]
```diff
@@ -13610,6 +13610,12 @@ BlurredLocation = function BlurredLocation(options) {
   options.gridSystem = options.gridSystem || require('./core/gridSystem.js');
   options.Interface = options.Interface || require('./ui/Interface.js');
 
+  options.Geocode = options.Geocode || function Geocode(geometry) {
+    $("#lat").val(geometry.lat);
```
Comment (jywarren) [inline dist/Leaflet.BlurredLocation.js]: Oh actually let's move this down into the onComplete method too!

Comment (mridulnagpal) [inline dist/Leaflet.BlurredLocation.js]: Done

**Comment by jywarren** [comment]
Ok then, let's merge!

**Comment by mridulnagpal** [comment]
Cheers!

**Pull request merged**